### ISSUE TITLE
how to using WARP proxy for outbound like xray core?

### ISSUE THREAD

**Issue by beskooitpedar** [p4gefau1t/trojan-go#508]
Hi. I get a Google 403 error on my servers. problem solved by using WARP proxy in this solution:
https://github.com/Musixal/Marzban-Tools/tree/main/WARP
in Xray core, we have an outbound section to send traffic through it like the picture:
https://user-images.githubusercontent.com/124933548/229277954-17cfa0e8-6997-4510-9ab3-1f77f23b8185.png
what we should do in Trojan Go?

**Comment by CXwudi** [comment]
Use forwarding https://p4gefau1t.github.io/trojan-go/advance/forward/

**Comment by lidotcircle** [comment]
you can use `forward_proxy` in config file

``` json
{
  "run_type": "server",
...
  "forward_proxy": {
    "enabled": true,
    "proxy_addr": "<socks5-address>",
    "proxy_port": "<socks5-port>",
    "username": "<optional>",
    "password": "<optional>"
  }
}
```

**Comment by beskooitpedar** [comment]
tnx.any way to proxy a special site with proxy from geo file?
example:
proxy google with warp but other sites use a server ip...

**Comment by luokuning** [comment]
配置了 `forward_proxy` 错误会出现以下错误：
```
[ERROR] 2024/10/05 08:49:48 github.com/p4gefau1t/trojan-go/proxy.(*Proxy).relayConnLoop.func1.1:proxy.go:66 proxy failed to dial connection | freedom failed to dial target address via socks proxy ontology.health.apple.com:443 | socks connect tcp 127.0.0.1:40000->ontology.health.apple.com:443: unexpected EOF
```

我能确认 warp 是正确运行的，状态是 Connected，并且运行 `curl -x socks5://127.0.0.1:40000 https://www.google.com` 能正确返回。

**Comment by krfantasy** [comment]
> 配置了 `forward_proxy` 错误会出现以下错误：
> 
> ```
> [ERROR] 2024/10/05 08:49:48 github.com/p4gefau1t/trojan-go/proxy.(*Proxy).relayConnLoop.func1.1:proxy.go:66 proxy failed to dial connection | freedom failed to dial target address via socks proxy ontology.health.apple.com:443 | socks connect tcp 127.0.0.1:40000->ontology.health.apple.com:443: unexpected EOF
> ```
> 
> 我能确认 warp 是正确运行的，状态是 Connected，并且运行 `curl -x socks5://127.0.0.1:40000 https://www.google.com` 能正确返回。

I have the same issue after upgrading `cloudflare-warp`

**Comment by moooyo** [comment]
@krfantasy  Did you resolve this issue? I have the same issue.

**Comment by moooyo** [comment]
@krfantasy I've been trying to debug this issue. But it seems not relate to any third-party package. And seems that it's cloudflare's bug. Finally I downgrade cloudflare-warp and this issue has been resolved.

**Comment by krfantasy** [comment]
@moooyo Yes, downgrading fixed the problem. It's a bug in the latest version `cloudflare-warp` proxy mode,
https://community.cloudflare.com/t/cloudflare-warp-version-2024-9-346-0-proxy-mode-issue/720751.

The only thing we can do is downgrade to version `2024.6.497-1` or wait for cloudflare to fix their bug.
